src/PilatusCamera.cpp

```diff
@@ -296,6 +298,41 @@ void Camera::_resync()
     send("version");
 }
 
+//-----------------------------------------------------
+//
+//-----------------------------------------------------
+void Camera::_checkcmd()
+{
+  DEB_MEMBER_FUNCT();
+  char messages[16384];
+  int aMessageSize = recv(m_socket,messages,sizeof(messages),0);
```

Comment:
This will conflicts with the recv in the thread of Camera::_run.

Reply:
Camera::_checkcmd is called in Camera::connect and it has a lock that prevents from communication conflicts.

Camera::_checkcmd justs reads the answers received from Camera::_resync calls.

PD1: In fact, it may be worth to move _resync from Camera::_connect to Camera::connect.
PD2: It may also be worth to merge _resync and _checkcmd, but this depends on it's use in _run and _reinit

Reply:
Ok, It's more clear for me now...
I can merge your code has it is or with the modification you suggest, it's up to you, just tell me.
But could you remove the code in Camera::_run which check the command availability? to avoid any mistake.  
Thanks again for your contribution.
P.S: You need to rebase on our master to fix conflict.

Reply:
Actually I've look a bit more deeper and I think it's safer to let the Camera::_run method do the socket reading because the _checkcmd may "eat" some other reply.
So to fix the bug you notice (Commands availability has to be checked when the camera is constructed), may I proposed you a patch (https://github.com/esrf-bliss/Lima-camera-pilatus/tree/wait_checkcmd). I can't tested it now, but the idea is to synchronize the connection with the reply of  roi and setenergy command.
Tell me if it suit your need.

SEB